### PR TITLE
Right Click Interact

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Done

### CODE REVIEW

File: src/OpenLoco/Input/MouseInput.cpp [hunk 1]
```diff
@@ -648,11 +649,13 @@ namespace OpenLoco::Input
...
-        registers regs{};
-        regs.edx = reinterpret_cast<uint32_t>(tree);
-        regs.ax = pos.x;
-        regs.cx = pos.y;
-        call(0x004BB116, regs);
+        GameCommands::setErrorTitle(StringIds::error_cant_remove_this);
+        GameCommands::TreeRemovalArgs args;
+        args.pos = pos;
+        args.baseZ = tree->baseZ();
+        args.elementType = tree->rawData()[0];
```
Comment (duncanspumpkin) [inline src/OpenLoco/Input/MouseInput.cpp:656]: annoyingly i dont have enough information to accurately split this up into its component parts atm. I think only rotation (first 3 bits) is unique but as I am unsure the whole byte is being sent.

File: src/OpenLoco/Input/MouseInput.cpp [hunk 2]
```diff
@@ -590,99 +591,166 @@ namespace OpenLoco::Input
...
-        call(0x0042D9BF, regs);
+        GameCommands::setErrorTitle(StringIds::error_cant_remove_this);
+        GameCommands::BuildingRemovalArgs args;
+        Pos2 firstTile = pos - Map::offsets[building->multiTileIndex()];
+        args.pos = Pos3(firstTile.x, firstTile.y, building->baseZ() * 4);
+        GameCommands::do_45(GameCommands::Flags::apply, args);
     }
 
     // 0x004C4809 TODO: Move to a better file
     static void wallInteract(Map::WallElement* wall, const Map::Pos2 pos)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Input/MouseInput.cpp:737]: How is this triggered? I find myself unable to right-click walls/fences to delete them.

File: src/OpenLoco/Input/MouseInput.cpp [hunk 3]
```diff
@@ -590,99 +591,166 @@ namespace OpenLoco::Input
...
-        call(0x004C4809, regs);
+        GameCommands::setErrorTitle(StringIds::error_cant_remove_this);
+        GameCommands::WallRemovalArgs args;
+        args.pos = Pos3(pos.x, pos.y, wall->baseZ() * 4);
+        args.rotation = wall->rotation();
+        GameCommands::do_33(GameCommands::Flags::apply, args);
     }
 
     // 0x0042F007 TODO: Move to a better file
     static void headquarterInteract(Map::BuildingElement* building, const Map::Pos2 pos)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Input/MouseInput.cpp:747]: How is this triggered? I can't remove HQs by right-clicking them. (Not on master either)

**Review by AaronVanGeffen — APPROVED** [review]
Code looks fine. Tested most functions to work correctly; unsure about fences and HQs.

File: src/OpenLoco/Input/MouseInput.cpp (hunk 3 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Input/MouseInput.cpp:747]: You can trigger them if you have a window open like the construction window.

File: src/OpenLoco/Input/MouseInput.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Input/MouseInput.cpp:737]: I have not been able to trigger this. The wall is excluded from the filter so I don't think this is reachable atm.

**Pull request merged**